src/cascadia/TerminalSettingsEditor/Appearances.cpp

```diff
@@ -42,6 +43,145 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
         return _hasPowerlineCharacters.value_or(false);
     }
 
+    Windows::Foundation::Collections::IMap<winrt::hstring, winrt::hstring> Font::FontAxesTagsAndNames()
+    {
+        auto fontAxesTagsAndNames = winrt::single_threaded_map<winrt::hstring, winrt::hstring>();
+
+        wil::com_ptr<IDWriteFont> font;
+        THROW_IF_FAILED(_family->GetFont(0, font.put()));
+        wil::com_ptr<IDWriteFontFace> fontFace;
+        THROW_IF_FAILED(font->CreateFontFace(fontFace.put()));
+        wil::com_ptr<IDWriteFontFace5> fontFace5;
+        if (fontFace5 = fontFace.try_query<IDWriteFontFace5>())
+        {
+            wil::com_ptr<IDWriteFontResource> fontResource;
+            THROW_IF_FAILED(fontFace5->GetFontResource(fontResource.put()));
+
+            std::vector<DWRITE_FONT_AXIS_VALUE> axesVector;
+            const auto axesCount = fontFace5->GetFontAxisValueCount();
+            if (axesCount > 0)
+            {
+                uint32_t localeIndex;
+                BOOL localeExists;
+                wchar_t localeName[LOCALE_NAME_MAX_LENGTH];
+                const auto localeToTry = GetUserDefaultLocaleName(localeName, LOCALE_NAME_MAX_LENGTH) ? localeName : L"en-US";
+
+                axesVector.resize(axesCount);
+                fontFace5->GetFontAxisValues(axesVector.data(), axesCount);
+                for (uint32_t i = 0; i < axesCount; ++i)
+                {
+                    const auto tagString = _axisTagToString(axesVector[i].axisTag);
+
+                    wil::com_ptr<IDWriteLocalizedStrings> names;
+                    THROW_IF_FAILED(fontResource->GetAxisNames(i, names.put()));
+
+                    if (!SUCCEEDED(names->FindLocaleName(localeToTry, &localeIndex, &localeExists)) || !localeExists)
+                    {
+                        // default to the first locale in the list
+                        localeIndex = 0;
+                    }
+
+                    uint32_t length;
+                    if (SUCCEEDED(names->GetStringLength(localeIndex, &length)))
+                    {
+                        // it is reasonable to assume that the name length is not going to exceed 512 chars
+                        wchar_t name[512];
+                        if (SUCCEEDED(names->GetString(localeIndex, name, length + 1)))
+                        {
+                            fontAxesTagsAndNames.Insert(tagString, winrt::hstring{ name });
+                            continue;
+                        }
+                    }
+                    // if there was no name found, it means the font does not actually support this axis
+                    // don't insert anything into the vector in this case
+                }
+            }
+        }
+        return fontAxesTagsAndNames;
+    }
+
+    winrt::hstring Font::_axisTagToString(DWRITE_FONT_AXIS_TAG tag)
+    {
+        std::wstring result;
+        for (int i = 0; i < 4; ++i)
+        {
+            result.push_back((tag >> (i * 8)) & 0xFF);
```

Comment:
note to self:
> ```DWRITE_MAKE_FONT_AXIS_TAG('c', 's', 't', 'm');```

so this is literally iterating over each of the `char`s embedded in the value of the `DWRITE_FONT_AXIS_TAG` and then sticking them into a string